extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -233,5 +236,191 @@ export class ProjectsController {
 		return treeNode instanceof FolderNode ? utils.trimUri(treeNode.root.uri, treeNode.uri) : '';
 	}
 
+	/**
+	 * Imports a new SQL database project from the existing database,
+	 * prompting the user for a name, file path location and extract target
+	 */
+	public async importNewDatabaseProject(context: any): Promise<void> {
+		let model = <ImportDataModel>{};
+
+		try {
+			let profile = context ? <IConnectionProfile>context.connectionProfile : undefined;
+			//TODO: Prompt for new connection addition and get database information if context information isn't provided.
+			if (profile) {
+				model.serverId = profile.id;
+				model.database = profile.databaseName;
+			}
+
+			// Get project name
+			let newProjName = await this.getProjectName(model.database);
+
+			if (!newProjName) {
+				this.apiWrapper.showErrorMessage(constants.projectNameRequired);
+				return;
+			}
+			model.projName = newProjName;
+
+			// Get extractTarget
+			let extractTarget: ExtractTarget = await this.getExtractTarget();
+			model.extractTarget = extractTarget;
+
+			// Get folder location for project creation
+			let newProjUri = await this.getFolderLocation(model.extractTarget);
+			if (!newProjUri) {
+				this.apiWrapper.showErrorMessage(constants.projectLocationRequired);
+				return;
+			}
+
+			let newProjFolderUri;
+			if (extractTarget !== 1) {
+				newProjFolderUri = newProjUri;
+			} else {
+				// Get filename, if extractTarget = File
+				newProjFolderUri = utils.trimFileName(newProjUri);
+			}
+
+			// TODO: what if the selected folder is outside the workspace?
+			model.filePath = newProjUri.fsPath;
+
+			//Set model version
+			model.version = '1.0.0.0';
+
+			// Call ExtractAPI in DacFx Service
+			await this.importApiCall(model);
+			// TODO: Check for success
+
+			// Create and open new project
+			const newProjFilePath = await this.createNewProject(newProjName as string, newProjFolderUri as Uri);
+			const project = await this.openProject(Uri.file(newProjFilePath));
+
+			//Create a list of all the files and directories to be added to project
+			let fileFolderList: string[] = await this.generateList(model.filePath);
+
+			// Add generated file structure to the project
+			await project.addToProject(fileFolderList);
+
+			//Refresh project to show the added files
+			this.refreshProjectsTree();
+		}
+		catch (err) {
+			this.apiWrapper.showErrorMessage(utils.getErrorMessage(err));
+		}
+	}
+
+	private async getProjectName(dbName: string): Promise<string | undefined> {
+		let projName = await this.apiWrapper.showInputBox({
+			prompt: constants.newDatabaseProjectName,
+			value: `DatabaseProject${dbName}`
+		});
+
+		return projName;
+	}
+
+	private async getExtractTarget(): Promise<ExtractTarget> {
+		let extractTarget: ExtractTarget;
+
+		// eslint-disable-next-line code-no-unexternalized-strings
+		let extractTargetOptions: string[] = Object.keys(ExtractTarget).filter(k => typeof ExtractTarget[k as any] === "number");
```

Comment:
shouldn't be disabling eslint. This error will go away if single quotes are used.